src/MessagePack.UnityClient/Assets/Scripts/MessagePack/SequencePool.cs

```diff
@@ -94,9 +105,13 @@ internal Rental Rent()
                 }
             }
 
+            var sequence = this.arrayPoolOrMemoryPool is ArrayPool<byte> arrayPool
+                ? new Sequence<byte>(arrayPool) { MinimumSpanLength = MinimumSpanLength }
+                : new Sequence<byte>((MemoryPool<byte>)this.arrayPoolOrMemoryPool) { MinimumSpanLength = MinimumSpanLength };
```

Comment:
Can you avoid the repetition of initializing properties by switching from an object initializer to just setting the property once after you construct using either constructor?